File: api_nestjs_distributed_order_platform/src/presentation/http/mappers/order.mapper.ts
import { OrderEntity } from "src/domain/entities/order.entity";

export function toOrderResponse(order: OrderEntity) {
    return {
        id: order.getId(),
        clientId: order.getClientId(),
        status: order.getStatus(),
        total: order.getTotal(),
        version: order.getVersion(),
        items: order.getItems().map((item) => ({
            productId: item.getProductId(),
            quantity: item.getQuantity(),
            unitPrice: item.getUnitPrice(),
            subtotal: item.getSubTotal(),
        })),
    };
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/presentation/http/mapper/OrderMapper.java
package com.project.order.presentation.http.mapper;

import com.project.order.domain.model.Order;
import com.project.order.presentation.http.dto.OrderDto.OrderItemResponse;
import com.project.order.presentation.http.dto.OrderDto.OrderResponse;

public class OrderMapper {

    private OrderMapper() {}

    public static OrderResponse toResponse(Order order) {
        return new OrderResponse(
                order.getId(),
                order.getClientId(),
                order.getStatus().name(),
                order.getTotal().getValue(),
                order.getVersion(),
                order.getItems().stream()
                        .map(item -> new OrderItemResponse(
                                item.getProductId(),
                                item.getQuantity(),
                                item.getUnitPrice().getValue(),
                                item.getSubTotal().getValue()
                        ))
                        .toList()
        );
    }
}
